beginner_source/blitz/autograd_tutorial.py

```diff
@@ -1,198 +1,276 @@
+#       \frac{\partial y_{1}}{\partial x_{n}} & \cdots & \frac{\partial y_{m}}{\partial x_{n}}
+#       \end{array}\right)\left(\begin{array}{c}
+#       \frac{\partial l}{\partial y_{1}}\\
+#       \vdots\\
+#       \frac{\partial l}{\partial y_{m}}
+#       \end{array}\right)=\left(\begin{array}{c}
+#       \frac{\partial l}{\partial x_{1}}\\
+#       \vdots\\
+#       \frac{\partial l}{\partial x_{n}}
+#       \end{array}\right)
+#
+# This characteristic of vector-Jacobian product is what we use in the above example;
+# ``external_grad`` represents :math:`\vec{v}`.
+#
+
+
+
+######################################################################
+# Computational Graph
+# ~~~~~~~~~~~~~~~~~~~
+#
+# Conceptually, autograd keeps a record of data (tensors) & all executed
+# operations (along with the resulting new tensors) in a directed acyclic
+# graph (DAG) consisting of
+# `Function <https://pytorch.org/docs/stable/autograd.html#torch.autograd.Function>`__
+# objects. In this DAG, leaves are the input tensors, roots are the output
+# tensors. By tracing this graph from roots to leaves, you can
+# automatically compute the gradients using the chain rule.
+#
+# In a forward pass, autograd does two things simultaneously: \* run the
+# requested operation to compute a resulting tensor, and \* maintain the
+# operation’s *gradient function* in the DAG. This is stored in the
+# resulting tensor’s .\ ``grad_fn`` attribute.
+#
+# The backward pass kicks off when ``.backward()`` is called on the DAG
+# root. Autograd then \* computes the gradients from each ``.grad_fn``, \*
+# accumulates them in the respective tensor’s ``.grad`` attribute, and \*
+# using the chain rule, propagates all the way to the leaf tensors.
+#
+# .. Note::
+#   **Autograd DAGs are dynamic in PyTorch**
+#   An important thing to note is that the graph is recreated from scratch; after each
+#   ``.backward()`` call, autograd starts populating a new graph. This is
+#   exactly what allows you to use control flow statements in your model;
+#   you can change the shape, size and operations at every iteration if
+#   needed. Autograd does not need you to encode all possible paths
+#   beforehand.
+#
+
+
+######################################################################
+# Exclusion from the DAG
+# ^^^^^^^^^^^^^^^^^^^^^^
+#
+# Autograd tracks operations on all tensors which have their
+# ``requires_grad`` flag set to ``True``. For tensors that don’t require
+# gradients, setting this attribute to ``False`` excludes it from the
+# gradient computation DAG and increases efficiency.
+#
+# The output tensor of an operation will require gradients even if only a
+# single input tensor has ``requires_grad=True``.
+#
+
+x = torch.rand(5, 5)
+y = torch.rand(5, 5)
+z = torch.rand((5, 5), requires_grad=True)
+
+a = x + y
+print("Does `a` require gradients?")
+print(a.requires_grad==True)
+b = x + z
+print("Does `b` require gradients?")
+print(b.requires_grad==True)
```

Comment:
would it be helpful to show the output here?

Also would it be sufficient to simply say `print(a.requires_grad)` ?